packages/svelte/src/compiler/phases/3-transform/css/index.js

```diff
@@ -209,20 +257,54 @@ const visitors = {
 
 					break;
 				}
-				first = false;
 			}
 		}
 
 		context.next();
+	},
+	PseudoClassSelector(node, context) {
+		if (node.name === 'is' || node.name === 'where') {
+			context.next();
+		}
 	}
 };
 
 /** @param {import('#compiler').Css.Rule} rule */
 function is_empty(rule) {
-	if (rule.block.children.length > 0) return false;
+	for (const child of rule.block.children) {
+		if (child.type === 'Declaration') {
+			return false;
+		}
+
+		if (child.type === 'Rule') {
+			if (is_used(child) && !is_empty(child)) return false;
+		}
+
+		if (child.type === 'Atrule') {
+			return false; // TODO
+		}
+	}
+
 	return true;
 }
 
+/** @param {import('#compiler').Css.Rule} rule */
+function is_used(rule) {
+	for (const selector of rule.prelude.children) {
+		if (selector.metadata.used) return true;
+	}
+
+	for (const child of rule.block.children) {
+		if (child.type === 'Rule' && is_used(child)) return true;
+
+		if (child.type === 'Atrule') {
+			return true; // TODO
```

Comment:
What does this TODO mean?

Reply:
that we don't yet analyse atrules to determine whether they are used in the context of the current component. it was there before this PR but in a different place